.circleci/config.yml

```diff
@@ -61,6 +81,8 @@ jobs:
 
   build_debug:
     <<: *android_config
+    environment:
+        GRADLE_OPTS: '-Dorg.gradle.jvmargs="-Xms128m -Xmx1024m -XX:+HeapDumpOnOutOfMemoryError" -DpreDexEnable=false'
```

Comment:
A comment here about how you reached these magic flags would be nice.

Reply:
Here it is, I'll add it as comment in the source too:
       Runtime configuration build tests
       ./gradlew clean assembleDebug
       Xms128 - the default on most systems is 256m but gradle spawns some java subprocesses during build that don't need 256m
       Xmx512m 1m30s
       Xmx768m 1m17s
       Xmx1024m ~1min
       Xmx1536m ~1min
       -XX:+HeapDumpOnOutOfMemoryError in case of an out of memory the java process will dump a heap file on the file system.
       The heap file can be analyzed to figure out the root cause or the memory leak.
       -DpreDexEnable=false disables pre-dexing step. Pre-dexing is good for local incremental builds during development,
       But on a build machine, the performance impact is not worthy because on a CI, every build should be clean.